src/apps/front-office/design-system/layouts/Header/MiddleHeader.tsx

```diff
@@ -0,0 +1,120 @@
+import { Link } from "@mongez/react-router";
+// import { useState } from "react";
+import { useState } from "react";
+import logo from "shared/assets/images/logo.svg";
+
+import { Input } from "../../components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../../components/ui/select";
+import { Separator } from "../../components/ui/separator";
+import { useWindowScroll } from "../../hooks";
+import {
+  MiddleHeaderInputPlacholder,
+  MiddleHeaderSelectPlacholder,
+  middleHeaderActions,
+  middleHeaderCategories,
+  navbarIcons,
+} from "./constant/middleHeaderData";
+import AccountMenu from "./menu/AccountMenu";
+import CartMenu from "./menu/CartMenu";
+import NavigationMenu from "./menu/NavigationMenu";
+
+function capitalize(value: string): string {
+  return value.charAt(0).toUpperCase() + value.slice(1);
+}
+
+const MiddleHeader = () => {
+  const windowScroll = useWindowScroll();
+  const [openNavMenu, setOpenNavMenu] = useState(false);
```

Comment:
I'd recommend using [useBooleanState hook](https://github.com/hassanzohdy/react-hooks#usebooleanstate) to  reduce code